openex-api/src/main/java/io/openex/service/UserService.java

```diff
@@ -28,86 +31,101 @@
-    }
-
-    public String encodeUserPassword(String password) {
-        return passwordEncoder.encode(password);
+  private final Argon2PasswordEncoder passwordEncoder = new Argon2PasswordEncoder();
+  private UserRepository userRepository;
+  private TokenRepository tokenRepository;
+  private TagRepository tagRepository;
+  private GroupRepository groupRepository;
+  private OrganizationRepository organizationRepository;
+
+  @Autowired
+  public void setOrganizationRepository(OrganizationRepository organizationRepository) {
+    this.organizationRepository = organizationRepository;
+  }
+
+  @Autowired
+  public void setTagRepository(TagRepository tagRepository) {
+    this.tagRepository = tagRepository;
+  }
+
+  @Autowired
+  public void setGroupRepository(GroupRepository groupRepository) {
+    this.groupRepository = groupRepository;
+  }
+
+  @Autowired
+  public void setUserRepository(UserRepository userRepository) {
+    this.userRepository = userRepository;
+  }
+
+  @Autowired
+  public void setTokenRepository(TokenRepository tokenRepository) {
+    this.tokenRepository = tokenRepository;
+  }
+
+  // region users
+  public boolean isUserPasswordValid(User user, String password) {
+    return passwordEncoder.matches(password, user.getPassword());
+  }
+
+  public void createUserSession(User user) {
+    List<SimpleGrantedAuthority> roles = new ArrayList<>();
+    roles.add(new SimpleGrantedAuthority(ROLE_USER));
+    if (user.isAdmin()) {
+      roles.add(new SimpleGrantedAuthority(ROLE_ADMIN));
     }
-
-    public void createUserToken(User user) {
-        Token token = new Token();
-        token.setUser(user);
-        token.setCreated(now());
-        token.setValue(UUID.randomUUID().toString());
-        tokenRepository.save(token);
-    }
-
-    public User updateUser(User user) {
-        return userRepository.save(user);
-    }
-
-    public User createUser(CreateUserInput input, int status) {
-        User user = new User();
-        user.setUpdateAttributes(input);
-        user.setStatus((short) status);
-        if (StringUtils.hasLength(input.getPassword())) {
-            user.setPassword(encodeUserPassword(input.getPassword()));
-        }
-        user.setTags(fromIterable(tagRepository.findAllById(input.getTagIds())));
-        user.setOrganization(updateRelation(input.getOrganizationId(), user.getOrganization(), organizationRepository));
-        // Find automatic groups to assign
-        List<Group> assignableGroups = groupRepository.findAll(GroupSpecification.defaultUserAssignable());
-        user.setGroups(assignableGroups);
-        // Save the user
-        User savedUser = userRepository.save(user);
-        createUserToken(savedUser);
-        return savedUser;
+    SecurityContext context = SecurityContextHolder.createEmptyContext();
+    context.setAuthentication(new PreAuthenticatedAuthenticationToken(new OpenexPrincipal() {
+      @Override
+      public String getId() {
+        return user.getId();
```

Comment:
It seems a bit strange to me to have methods based on objects outside the created class